src/pages/ProblemEditor/ProblemEditorContainer/ProblemEditorNavigator.tsx

```diff
@@ -0,0 +1,88 @@
+import { Button, Box } from "@mui/material";
+import React from "react";
+import { useDispatch } from "react-redux";
+import { useAppSelector } from "../../../app/common/hooks";
+import {
+  incrementActiveStep,
+  decrementActiveStep,
+} from "./problemEditorContainerSlice";
+
+interface Props {
+  formRef: any;
+}
+
+export const ProblemEditorNavigator = ({ formRef }: Props) => {
+  const dispatch = useDispatch();
+  const activeStep = useAppSelector(
+    (state) => state.problemEditorContainer.activeStep
+  );
+
+  //TODO put this logic in the reducer
+  /* STEP SELECTORS */
+  const metadataIsValid = useAppSelector(
```

Comment:
I think you could probably get all of these values with something like:
const {metadataIsValid, problemIsValid, codeIsValid, serverConfigIsValid, testEditorIsValid } = use AppSelector((state) => state.problemEditorContainer);